mllib-local/src/main/scala/org/apache/spark/ml/linalg/Vectors.scala

```diff
@@ -186,6 +186,46 @@ sealed trait Vector extends Serializable {
    */
   @Since("3.0.0")
   def dot(v: Vector): Double = BLAS.dot(this, v)
+
+  /**
```

Comment:
what about also adding `def +(a: Double)`, `def -(a: Double)` for scalar input and `def *(v: Vector)`, `def /(v: Vector)` as element-wise ops?

Reply:
Seems reasonable. I'd generally prefer to focus on parity with Python; is that what __add__ supports for example?

Reply:
I am not a python expert by any means, but I think this type of behavior is automatically supported with [broadcasting](https://docs.scipy.org/doc/numpy/reference/generated/numpy.add.html)?  @zhengruifeng Can you verify?

If so I am happy to implement.

Reply:
Sorry for late reply.
Yes numpy support broadcasting.
But I guess what @srowen means is those `Vector`/`Vectors` class defined in 'spark/python/pyspark/ml/linalg/__init__.py'